go/obscuronode/enclave/state.go

```diff
@@ -109,33 +111,34 @@ func updateState(b *types.Block, s Storage, blockResolver BlockResolver) *blockS
 	}
 
 	rollups := extractRollups(b, blockResolver)
-
 	genesisRollup := s.FetchGenesisRollup()
-	if len(rollups) == 1 {
-		rol := rollups[0]
-
-		// the incoming block might hold the genesis rollup
-		// todo change this to an hardcoded hash on testnet/mainnet
-		if genesisRollup == nil && rol.Header.Height == obscurocommon.L2GenesisHeight {
-			s.StoreGenesisRollup(rol)
-			genesisRollup = rol
-		}
+
+	// processing blocks before genesis, so there is nothing to do
+	if genesisRollup == nil && len(rollups) == 0 {
+		return nil
+	}
+
+	// the incoming block holds the genesis rollup
+	// todo change this to an hardcoded hash on testnet/mainnet
+	if genesisRollup == nil && len(rollups) == 1 {
```

Comment:
how could that be? The genesis has to be the first ( height=0)

Reply:
This will take any rollup that it finds as a genesis, suppose that's fine though.. when mainnet arrives we hardcode the hash and establish the canonical chain 👍

Reply:
ah, I see what you mean. Like a sanity check to make sure the "height" property in the header is set to 0. 

This is not the "final" solution for this anyway. I guess, the next stuff you're doing with the l1 integration will change it further anyway.